### PR TITLE
Add JAX versions of shallow_water_simple.py and shallow_water_nonlinear.py

### PR DESCRIPTION
**Pull request by lobpcg**
JAX versions shallow_water_simple_jax.py of shallow_water_simple.py and shallow_water_nonlinear_jax.py of shallow_water_nonlinear.py together with pytest modulus to test the accuracy.

### CODE REVIEW

**Comment by lobpcg** [comment]
? @dionhaefner PRs are not welcome?
